test/versioned/node-rdkafka/kafka.tap.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+const { removeModules } = require('../../lib/cache-buster')
+
+tap.beforeEach(async (t) => {
+  t.context.agent = helper.instrumentMockedAgent()
+
+  const Kafka = require('node-rdkafka')
+  t.context.Kafka = Kafka
+
+  await new Promise((resolve) => {
+    const producer = new Kafka.Producer({
+      'metadata.broker.list': '127.0.0.1:9092'
+    })
+    producer.connect()
+    producer.setPollInterval(10)
+    producer.on('ready', () => {
+      t.context.producer = producer
+      resolve()
+    })
+  })
+
+  await new Promise((resolve) => {
+    const consumer = new Kafka.KafkaConsumer({
+      'metadata.broker.list': '127.0.0.1:9092',
+      'group.id': 'kafka'
+    })
+    consumer.connect()
+    consumer.on('ready', () => {
+      t.context.consumer = consumer
+      resolve()
+    })
+  })
+})
+
+tap.afterEach(async (t) => {
+  helper.unloadAgent(t.context.agent)
+  removeModules(['node-rdkafka'])
+
+  await new Promise((resolve) => {
+    t.context.producer.disconnect(resolve)
+  })
+  await new Promise((resolve) => {
+    t.context.consumer.disconnect(resolve)
+  })
+})
+
+tap.test('stub', { timeout: 10_000 }, (t) => {
```

Comment:
I removed this timeout and it works as expected

Reply:
The timeout ensures that the test will get terminated and the cleanup actions will be performed when running the script directly (or other odd circumstances).